app/models/concerns/github_identity.rb

```diff
@@ -34,9 +34,21 @@ def update_repo_permissions_async
     SyncPermissionsWorker.perform_async(self.id)
   end
 
+  def update_auth_token
+    at = AuthToken.find_or_create_by(login: nickname)
+    at.token = token
+    if at.still_authorized?
```

Comment:
I think my only concern, besides not really knowing what other code would be affected by this, is `still_authorized?` on `AuthToken` could return false if the token has hit its rate limit at the time this method is run and would be deleted when running the rake task. I don't know if that is really a valid concern since that is a very low chance of happening.